server/hedley/modules/custom/hedley_admin/scripts/generate-acute-illness-completion-report.php

```diff
@@ -0,0 +1,1052 @@
+    "AND field_muac_value IS NOT NULL",
+    "AND field_nutrition_signs_value IS NOT NULL",
+    "AND field_findings_signs_general_value IS NOT NULL AND field_findings_signs_respiratory_value IS NOT NULL",
+    "AND field_fever_period_value IS NOT NULL",
+    "AND field_cough_period_value IS NOT NULL",
+    "AND field_nausea_period_value IS NOT NULL",
+    "AND field_treatment_history_value IS NOT NULL",
+  ];
+
+  $modifier = 0;
+
+  foreach ($constraints as $constraint_clause) {
+    $at_least_one = FALSE;
+
+    $results = db_query("SELECT DISTINCT field_acute_illness_encounter_target_id
+    FROM field_data_field_acute_illness_encounter e
+    LEFT JOIN node ON e.entity_id=node.nid
+    LEFT JOIN field_data_field_individual_participant ip ON e.field_acute_illness_encounter_target_id=ip.entity_id
+    LEFT JOIN field_data_field_person person ON ip.field_individual_participant_target_id=person.entity_id
+    LEFT JOIN field_data_field_health_center hc ON person.field_person_target_id=hc.entity_id
+    LEFT JOIN field_data_field_district district ON person.field_person_target_id=district.entity_id
+    LEFT JOIN field_data_field_ai_encounter_type t ON e.field_acute_illness_encounter_target_id=t.entity_id
+    LEFT JOIN field_data_field_heart hea ON node.nid=hea.entity_id
+    LEFT JOIN field_data_field_lungs lun ON node.nid=lun.entity_id
+    LEFT JOIN field_data_field_findings_signs_general gen ON node.nid=gen.entity_id
+    LEFT JOIN field_data_field_findings_signs_respiratory res ON node.nid=res.entity_id
+    LEFT JOIN field_data_field_heart_rate her ON node.nid=her.entity_id
+    LEFT JOIN field_data_field_respiratory_rate rer ON node.nid=rer.entity_id
+    LEFT JOIN field_data_field_body_temperature bod ON node.nid=bod.entity_id
+    LEFT JOIN field_data_field_dia dia ON node.nid=dia.entity_id
+    LEFT JOIN field_data_field_muac muac ON node.nid=muac.entity_id
+    LEFT JOIN field_data_field_nutrition_signs nutr ON node.nid=nutr.entity_id
+    LEFT JOIN field_data_field_fever_period fev ON node.nid=fev.entity_id
+    LEFT JOIN field_data_field_cough_period cou ON node.nid=cou.entity_id
+    LEFT JOIN field_data_field_nausea_period nau ON node.nid=nau.entity_id
+    LEFT JOIN field_data_field_travel_history tra ON node.nid=tra.entity_id
+    LEFT JOIN field_data_field_exposure exp ON node.nid=exp.entity_id
+    LEFT JOIN field_data_field_treatment_history tre ON e.entity_id=tre.entity_id
+    WHERE FROM_UNIXTIME(node.created) > :start_date
+      {$name_clause}
+      {$constraint_clause}
+      AND FROM_UNIXTIME(node.created) < :end_date",
+      [
+        ':start_date' => $start_date,
+        ':end_date' => $end_date,
+      ])->fetchAll(PDO::FETCH_COLUMN);
+
+    foreach ($results as $result) {
+      $ids[$result] += 1;
+      $at_least_one = TRUE;
+    }
+    if (!$at_least_one) {
+      $modifier += 1;
+    }
+
+  }
+
+  $count = 0;
+  foreach ($ids as $id) {
+    if ($id == count($constraints) - $modifier) {
```

Comment:
same as above, worth a comment, not trivial to follow the intention

Reply:
Hey Aron -  I added some explanatory comments to this section and the one mentioned above. Please let me know if you feel it could still benefit from a more detailed explanation or if it's good where it is.